### DISCUSSION TITLE
[Feature Request]: PDF meeting lists for website visitors

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#1056] [Ideas]
### Contact Details

os.website.a86@gmail.com

### Requested Feature/Enhancement

Hi team,

Thanks for all your great work on the meeting list plugin! I would not know what to do if your product did not exist!

Having said that, I would like to circle back to two features that have been discussed before. I am not sure what the status is.

**1. New design**
When I last tried out the new design on my staging site, I noticed that performing a search no longer includes the meeting notes field. Which means I can no longer do a front-end search by the unique WSO ID > the unique identifier that Al-Anon used for their groups. Has that changed? And if not, are there plans to create an additional field to enter that ID in and then to include this in the custom search in the list of meetings? It could be an optional field. I know that WSO ID and GSO ID are different phenomenons. But in Al-Anon, when it comes to the identity of a group, this WSO ID is the only fool proof way to find the right group.

**2. PDF lists for selected meetings or districts**
There used to be an option to create a PDF export for the admins. That is not what I am talking about. I know it has been removed. What I am referring to is a feature that website visitors can use to create a list of meetings for a specific region.

Example of how I envision this:
on my website I have a separate page with meeting lists for each District.
Each District is associated with s a list of regions, where the actual meetings are held.
So, in theory, if a website visitor would have the option to select a region, the underlying TSML code could match it with the list of regions and the relevant District. This could result in a PDF format list of meetings. It would not have to be a fancy list. Any kind of list would be helpful.

I know that some think that having such a PDF will result in printouts that are out of date the moment they are printed. This is true. But if members or newcomers find out that it is easy to reprint, they may not keep the list but just look up the PDF online. And not download and print it. And it would also help those who do not live on their computer, but like to have an old-fashioned piece of paper in their hand.

I know it will not be that simple to program for the TSML, but still. I would love to have that option. Volunteering to be a tester as well, if you would consider building such a feature.

Thanks for your consideration!
Nicky D

**Comment by tim-rohrer** [comment]
Hi Nicky. The idea of a front-side PDF generator was recently discussed. Not sure we're completely in favor of this yet. In the meantime, I suggest the admin generates the PDF and then puts a Wordpress link to it for end users. I realize this doesn't complete satisfy your request, but it is a viable workaround.

We strive to keep TSML responsive so that it should show reasonably well on mobile devices. I know a lot of people are using it that way, even AAers who have the Meeting Guide app available. 

We'll need to experiment with the search. I'm not aware of any changes but I could have missed something.

**Reply by tim-rohrer** [reply]
Nicky,

I've tested searching notes fields and it seems to work fine with the latest TSML. If you're experiencing something different, I suggest opening a separate Discussion topic.

Tim

**Comment by OSAwebsite** [comment]
Hi Tim,

Thanks for the quick response.
I still have my own lists right now, they are working well.
It's when the person that comes after me (end of this year) chooses to
fully rely in just the TSML tool that this will become an issue.

The lists are on this page, feel free to have a look on them. They are
based on the 'design' of the lists that existed on paper pre-Covid. When
Covid started, there were so many changes that printing the list was just
no use.

https://al-anon.alateen.on.ca/meeting-frontpage/meeting-prints/

I will respond by separate email to the new userinterface. I have not been
on the staging site for a while, but that is where I noticed it. Will check
in a moment and get back to you.




-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
I was not talking about the current user interface. I was talking about the
test new interface that we can toggle off and on. I just tested the TSML
interface and it seems to work now so I might change over to that design. I
spoke out of turn!

-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Reply by tim-rohrer** [reply]
I'm using TSML UI (the new interface) and it searches the fields just fine. If you've can take the time to test more, I'd to figure out why you're having difficulties with it.

What theme are you using? Other particulars like version numbers for Wordpress and TSML?

When you try searching, watch the debugging console on your browser, and see if any errors come up. Perhaps something is conflicting.

On a different but related topic, GitHub's email reply for their Discussion feature is HORRENDOUS, and if we could turn it off, we probably would. The problem is that it starts a new comment every time, so there is no thread development. Threads are much easier for future users to read and follow. So, if possible, it would be awesome if you could log into GitHub for your replies, and skip using the reply-by-email feature.

**Comment by OSAwebsite** [comment]
Unrelated question:
When I import a new meeting, I noticed that whatever is in the Notes field
(the field below the checkboxes in the back end) is stripped of hard
returns. Is there a way around this?


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
Tim, I have gone to Github for my topics. I thought that replying by email
would auto create a post and it would be just as easy. If it is not, I'll
go directly to Github.


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*
